tests/test_vsutil.py

```diff
@@ -161,3 +161,41 @@ def double_number(x: int) -> int:
 
         self.assertEqual(vsutil.iterate(2, double_number, 3), 16)
         self.assertEqual(vsutil.iterate(0, double_number, 4), 0)
+
+    def test_depth(self):  # TODO: test dither/range/range_in logic
```

Comment:
TODO for whom? Will you do that in a future PR? Why not now?

Reply:
The testing right now would require either a debug parameter in the function, or by using a hacky method with `sys.argv` and checking how the script was called (i.e. with pytest or unittest). As discussed on Discord, the hacky method is ugly and is only needed if we're trying to go for 100% code coverage again, but considering we don't even have a working CI testing system right now, there's really no need to in-depth test as I'm only using it here to make sure the function actually returns the expected clip/format. Another thing that might be better placed in another PR is testing this on actual example frames/clips to make sure the resize call (specifically the dithering) is working properly. I'd rather not bloat the repo with images though and would want to reliably host them somewhere else.

I'm also not even sure of a good way to test the range logic yet, so to answer `for whom` for this part, I guess whoever can find a good way to test this.